src/actions/ProfileActions.js

```diff
@@ -57,12 +70,13 @@ export const saveProfileFailure = error => ({
   payload: { error },
 });
 
-export const saveProfile = (username, userAccountState, fieldName) => ({
+export const saveProfile = (username, { profileData, preferencesData }, fieldName) => ({
   type: SAVE_PROFILE.BASE,
   payload: {
     fieldName,
     username,
-    userAccountState,
+    profileData,
```

Comment:
So this structure currently _works_ but is going to be refactored to be simpler.  Rather than passing around two separate structures for data, it'll all just be in one place as far as most of the application is concerned.